File: ManaDigital.Web/Models/Usuario.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ManaDigital.Web.Models;

[Table("usuarios")]
public class Usuario
{
    [Key]
    [Column("id")]
    public Guid Id {get; set;} = Guid.NewGuid();

    [Column("email")]
    [Required]
    public string Email {get; set;} = string.Empty;

    [Column("senha_hash")]
    [Required]
    public string SenhaHash {get; set;} = string.Empty;

    [Column("nome")]
    [Required]
    public string Nome {get; set;} = string.Empty;

    [Column("apelido")]
    public string Apelido {get; set;} = string.Empty;

    [Column("cargo")] 
    public string Cargo {get; set;} = "comum"; // 'comum' ou 'adm'

    [Column("pontos")]
    public int Pontos {get; set;} = 50;  // Começa com os 50 XP do bônus!

    [Column("convidado_por")]
    public string? ConvidadoPor { get; set; }

    [Column("data_criacao")]
    public DateTime DataCriacao {get; set;} = DateTime.UtcNow;

}

File: Mana_mobile/mana_app/lib/models/usuario.dart
class Usuario {
  final String id;
  final String nome;
  final String email;
  final int pontos;
  final String? apelido;
  final String? cargo;

  Usuario({
    required this.id,
    required this.nome,
    required this.email,
    required this.pontos,
    this.apelido,
    this.cargo,
  });

  Usuario copyWith({
    String? id,
    String? nome,
    String? email,
    int? pontos,
  }) {
    return Usuario(
      id: id ?? this.id,
      nome: nome ?? this.nome,
      email: email ?? this.email,
      pontos: pontos ?? this.pontos,
    );
  }

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      id: json['id'].toString(),
      nome: json['nome'] ?? '',
      apelido: json['apelido'],
      email: json['email'] ?? '',
      cargo: json['cargo'] ?? 'comum',
      pontos: json['pontos'] ?? 0,
    );
  }
}